src/retry.ts

```diff
@@ -0,0 +1,75 @@
+import * as retry from 'retry';
+import * as grpc from '@grpc/grpc-js';
+
+//default max retry value
+const FLIGHT_QUERY_MAX_RETRIES = 3; 
+const SPICE_NO_RETRY = '_SPICE_NO_RETRY';
+
+
+function dontRetry(err: any) {
```

Comment:
Agree custom error or even adding fn instead of flag which will be seen in logs/trace is a better option. Propose to keep as-is for now so we don't need to re-test the logic after adding changes. We will be able to improve this later if needed.